src/GraphQL/Extensions/TypeExtensions.cs

```diff
@@ -63,12 +64,67 @@ internal static bool IsNamedType(this Type type)
         }
 
         /// <summary>
-        /// Gets the GraphQL name of the type. This is derived from the type name and can be overridden by the <see cref="GraphQLMetadataAttribute"/>.
+        /// Gets the GraphQL name of the type. This is derived from the type name and can
+        /// be overridden by the <see cref="GraphQLMetadataAttribute"/>. The name is chosen
+        /// depending on <see cref="GlobalSwitches.UseLegacyTypeNaming"/>.
         /// </summary>
         /// <param name="type">The indicated type.</param>
         /// <returns>A string containing a GraphQL compatible type name.</returns>
         public static string GraphQLName(this Type type)
         {
+            if (!GlobalSwitches.UseLegacyTypeNaming)
+            {
+                return NameOf(type)
+                    .Replace('@', '_'); // F# anonymous class support
+
+                static string NameOf(Type type)
+                {
+                    type = type.GetNamedType();
+                    var attr = type.GetCustomAttribute<GraphQLMetadataAttribute>();
+                    if (!string.IsNullOrEmpty(attr?.Name))
+                    {
+                        return attr!.Name!;
```

Comment:
The `!` after `attr` is redundant.

Reply:
```suggestion
                        return attr.Name!;
```

Reply:
Unfortunately it doesn't work when targeting .NET Standard 2.0